File: domain/src/main/java/no/nav/sbl/dialogarena/sendsoknad/domain/Ektefelle.java
package no.nav.sbl.dialogarena.sendsoknad.domain;

import org.joda.time.LocalDate;

public class Ektefelle {
    private String navn;
    private LocalDate fodselsdato;
    private String fnr;
    private boolean folkeregistrertsammen;
    private boolean ikketilgangtilektefelle;

    public String getNavn() {
        return navn;
    }

    public LocalDate getFodselsdato() {
        return fodselsdato;
    }

    public String getFnr() {
        return fnr;
    }

    public boolean erFolkeregistrertsammen() {
        return folkeregistrertsammen;
    }

    public boolean harIkketilgangtilektefelle() {
        return ikketilgangtilektefelle;
    }

    public Ektefelle withNavn(String navn) {
        this.navn = navn;
        return this;
    }

    public Ektefelle withFodselsdato(LocalDate fodselsdato) {
        this.fodselsdato = fodselsdato;
        return this;
    }

    public Ektefelle withFnr(String fnr) {
        this.fnr = fnr;
        return this;
    }

    public Ektefelle withFolkeregistrertsammen(boolean folkeregistrertsammen) {
        this.folkeregistrertsammen = folkeregistrertsammen;
        return this;
    }

    public Ektefelle withIkketilgangtilektefelle(boolean ikketilgangtilektefelle) {
        this.ikketilgangtilektefelle = ikketilgangtilektefelle;
        return this;
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/personalia/person/domain/Ektefelle.kt
package no.nav.sosialhjelp.soknad.personalia.person.domain

import java.io.Serializable
import java.time.LocalDate

data class Ektefelle(
    val fornavn: String?,
    val mellomnavn: String?,
    val etternavn: String?,
    val fodselsdato: LocalDate?,
    val fnr: String?,
    val folkeregistrertSammen: Boolean,
    val ikkeTilgangTilEktefelle: Boolean
) : Serializable {
    constructor(ikkeTilgangTilEktefelle: Boolean) : this(null, null, null, null, null, false, ikkeTilgangTilEktefelle)
}
